textattack/shared/attack.py

```
@@ -166,6 +164,10 @@ def filter_transformations(
             current_text: The current ``AttackedText`` on which the transformation was applied.
             original_text: The original ``AttackedText`` from which the attack started.
         """
+        # Remove any occurences of current_text in transformed_texts
```

In which cases do we need to do this? I feel like this should be up to the transformation not to do?

---

I think this is a good idea. It's a bug that's appeared maybe 4 times in 4 separate transformations (most recently, BERT masked-LM swap) – and identity transformations break the search methods

---

Can we throw an error if there's a duplicate then? Rather than quietly filtering it out?

---

done